### PR TITLE
Remove Let's Encrypt variables, add aws_instance_profile

### PR DESCRIPTION
**Pull request by evanphx**

### CODE REVIEW

File: content/source/docs/enterprise/private/automating-the-installer.html.md [hunk 1]
```diff
@@ -71,9 +68,9 @@ The following apply to every installation:
   ```
 
 - `extern_vault_enable` - (Optional) Indicate if an external Vault cluster is being used. Set to `1` if so.
-  - `extern_vault_addr` - URL of external Vault cluster
-  - `extern_vault_role_id` - AppRole RoleId to use to authenticate with the Vault cluster
-  - `extern_vault_secret_id` - AppRole SecretId to use to authenticate with the Vault cluster
+  - `extern_vault_addr` - (Required) URL of external Vault cluster
+  - `extern_vault_role_id` - (Required) AppRole RoleId to use to authenticate with the Vault cluster
+  - `extern_vault_secret_id` - (Required) AppRole SecretId to use to authenticate with the Vault cluster
```
Comment (phinze) [inline content/source/docs/enterprise/private/automating-the-installer.html.md]: More like

> (Required if `extern_vault_enable` is `1`)

yeah?

Comment (evanphx) [reply content/source/docs/enterprise/private/automating-the-installer.html.md]: I'll make that clearer with a section bullet comment, as that fact is true for every indented variable there.

File: content/source/docs/enterprise/private/automating-the-installer.html.md [hunk 2]
```diff
@@ -71,9 +68,10 @@ The following apply to every installation:
   ```
 
 - `extern_vault_enable` - (Optional) Indicate if an external Vault cluster is being used. Set to `1` if so.
-  - `extern_vault_addr` - URL of external Vault cluster
-  - `extern_vault_role_id` - AppRole RoleId to use to authenticate with the Vault cluster
-  - `extern_vault_secret_id` - AppRole SecretId to use to authenticate with the Vault cluster
+  - These variables are only used if `extern_vault_enable` is set `1`
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/automating-the-installer.html.md]: I might go with "is set to"

File: content/source/docs/enterprise/private/automating-the-installer.html.md [hunk 3]
```diff
@@ -93,10 +91,15 @@ The following apply to every installation:
...
 
+Select which will be used, S3 or Azure:
+
+- `placement` - (Required) Set to `placement_s3` for S3 and `placement_azure` for Azure
+
 For S3:
 
-- `aws_access_key_id` — (Required) AWS access key ID for S3 bucket access. To use AWS instance profiles for this information, set it to `""`.
-- `aws_secret_access_key` — (Required) AWS secret access key for S3 bucket access. To use AWS instance profiles for this information, set it to `""`.
+- `aws_instance_profile` (Optional) When set, use credentials from the AWS instance profile. If selected, `aws_access_key_id` and `aws_secret_access_key` are not required. Set to `1` to use the instance profile. Defaults to `0`.
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/automating-the-installer.html.md]: I might move the "If selected" sentence to the end here.

File: content/source/docs/enterprise/private/automating-the-installer.html.md [hunk 4]
```diff
@@ -107,7 +110,7 @@ For Azure:
 - `azure_account_name` - (Required) The account name for the Azure account to access the container
 - `azure_account_key` - (Required) The account key to access the account specified in `azure_account_name`
 - `azure_container` - (Required) The identifer for the Azure blob storage container
-- `azure_endpoint` - (Optional) The URL for the Azure cluster to use. By default is the global one.
+- `azure_endpoint` - (Optional) The URL for the Azure cluster to use. By default this is the global one.
```
Comment (armchairlinguist) [inline content/source/docs/enterprise/private/automating-the-installer.html.md]: I think "global cluster" might be a bit clearer.

File: content/source/docs/enterprise/private/automating-the-installer.html.md (hunk 3 above)
Comment (evanphx) [reply content/source/docs/enterprise/private/automating-the-installer.html.md]: Not sure what you mean.

Comment (blalor) [reply content/source/docs/enterprise/private/automating-the-installer.html.md]: I think she's saying it should read like:

> When set, use credentials from the AWS instance profile.  Set to `1` to use the instance profile. Defaults to `0`.  If selected, `aws_access_key_id` and `aws_secret_access_key` are not required. 

I agree that reads a little easier.

File: content/source/docs/enterprise/private/automating-the-installer.html.md [hunk 5]
```diff
@@ -41,6 +41,65 @@ The settings file is JSON formatted. All values must be strings.  The example be
 }
 ```
 
+### Discovery
+
+One the easiest ways to get the settings is to [perform a manual install](./install-installer.html#installation) and configure all the settings how you want them. Then you can ssh in and request the settings in JSON format and use that file in a future automated install.
```
Comment (blalor) [inline content/source/docs/enterprise/private/automating-the-installer.html.md]: I dig this; it's really excellent.

**Pull request merged**